server.go

```diff
@@ -47,9 +47,21 @@ func (h *Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	query := r.URL.Query()
+	localPart := query.Get("l")
+
 	if strings.HasPrefix(path, "/hu/") {
 		hash := strings.TrimPrefix(path, "/hu/")
-		h.serveDiscovery(w, r, hash)
+		domain := strings.TrimPrefix(r.URL.Host, "openpgp.")
```

Comment:
This is the simple method, right? Then I don't think we should trim here. Otherwise it gives incorrect results for addresses like `root@openpgp.org`.

Reply:
Mmm, that should work . I mean to discover the key of *root@openpgp.org* the WKD client will do a query to `https://openpgp.openpgp.org/hu/<hash>?l=root`, so the `r.URL.Host` will be "openpgp.openpgp.org". `TrimPrefix` will only remove the first ocurrence of the prefix, so `strings.TrimPrefix("openpgp.openpgp.org", "openpgp.")` [will return "openpgp.org"](https://go.dev/play/p/RyT7Z1FA6br). Or am I missing something here?

Reply:
With the "simple method" described in the RFC, the domain name is used as-is, ie. no "openpgp" subdomain is used to make the request.

Reply:
I see, you are right. But if I understand correctly the RFC I don't need to trim anything, the simple method will always be directed to the original domain without additions, only the advanced method will be directed to a subdomain. I should pay more attention while reading RFCs. Thanks.

I'll remove the TrimPrefix.

Reply:
No worries :)